apps/yapms/src/lib/components/modals/toolsmodal/tools/TableModal.svelte

```diff
@@ -0,0 +1,128 @@
+<script lang="ts">
+	import { TableModalStore } from '$lib/stores/Modals';
+	import { RegionsStore } from '$lib/stores/regions/Regions';
+	import type { Region } from '$lib/types/Region';
+	import ModalBase from '../../ModalBase.svelte';
+	import { disableRegion, lockRegion } from '$lib/stores/regions/regionActions';
+	import { get } from 'svelte/store';
+	import CandidateActions from './tablecomponents/CandidateActions.svelte';
+
+	let filterInput = '';
+	$: isFiltering = filterInput !== '';
+	$: regionsToShow = filterRegions(filterInput);
+
+	function preventNonNumericalInput(e: KeyboardEvent) {
+		if (e.key !== 'Enter' && !e.key.match(/^[0-9]+$/)) e.preventDefault();
+	}
+
+	function preventNonNumericalPaste(e: ClipboardEvent) {
+		const pasteContents = e.clipboardData?.getData(e.clipboardData.types[0]);
+		if (!pasteContents?.match(/^[0-9]+$/)) e.preventDefault();
+	}
+
+	function updateRegionValue(
+		event: Event & { currentTarget: EventTarget & HTMLInputElement },
+		regionToUpdate: Region
+	) {
+		const newValue = Number(event.currentTarget.value);
+		const index = $RegionsStore.findIndex((region) => region.id === regionToUpdate.id);
+		if (!$RegionsStore[index].disabled) {
+			//Don't update value if disabled so the state stays disabled!
+			$RegionsStore[index].value = newValue;
+		}
+		$RegionsStore[index].permaVal = newValue;
+	}
+
```

Comment:
filterRegions below returns a map of the form (region id, boolean) where the bool determines whether or not to show that region in the table.

That is then used in this check:
`{#if !isFiltering || regionsToShow.get(region.id) == true}`

Where `~isFiltering` renders all regions if no filter is set.